src/main/kotlin/io/kuzzle/sdk/protocol/WebSocket.kt

```diff
@@ -114,13 +112,11 @@ open class WebSocket : AbstractProtocol {
             state = ProtocolState.OPEN
             trigger(NetworkStateChangeEvent(ProtocolState.OPEN))
 
-            thread(start = true) {
-                while (ws != null) {
-                    val payload = queue.poll()
-                    if (payload != null) {
-                        GlobalScope.launch {
-                            ws?.send(Frame.Text(payload))
-                        }
```

Comment:
By removing the thread logic you make the SDK synchrone which is not the purpose. You should maybe handle and clean created thread when reconnecting instead of make it synchrone